Day7/summary.js

```diff
@@ -0,0 +1,175 @@
+    setTimeout(() => {
+      resolve("Data Fetched succesfully By promises");
+    }, 1000);
+  });
+  
+  myPromise
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+  
+    console.log(" ")
+
+    // 4. Write a program to implement a Promise-based rate limiter, that limits the number of concurrent requests to a certain number
+
+const rateLimiter = (i) =>
+new Promise((resolve,reject) => {
+   if(i<10) resolve(`Promise run for ${i}th request`)
+   else
+    reject(`Too many concurrent request`)
+})
+
+for(let i=1;i<11;i++){
+rateLimiter(i).then((res) => console.log(res)).catch(err=> console.log(err))
+}
+
+
+console.log(" ")
+// 5.  Write a program to implement a Promise-based task queue, that processes tasks in a specified order, with a specified concurrency limit
+
+const taskQueue = (i,limit) =>
+  new Promise((resolve, reject) => {
+    setTimeout(() =>{
+    if (i <= limit) {
+        
+      resolve(`promise ran for ${i} requests`);
+    }
+     else {
+      reject("too many concurrent requests");
+    }
+},i*1000)}
+);
+    
+  
+const limit=10;
+for (let i = 1; i < limit+2; i++) {
+    taskQueue(i,limit)
+    .then((res) => {
+      console.log(res);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+console.log(" ")
+// 6. Implement a function that returns a resolved Promise after a specified delay using async/await.
+
+const myPromise1 = new Promise((resolve, reject) =>{
+    setTimeout(()=>{
+        resolve("Hello from the Promise");
+    },1000)
+})
+
+const getData = async () =>{
+    const promiseData = await myPromise1;
+    console.log(promiseData)
+}
+getData();
+
+console.log(" ")
+// 7. Create a function that performs multiple asynchronous operations in parallel using async/await and waits for all of them to complete before returning the results.
+
+const p1= new Promise((resolve, reject) =>{
+    setTimeout(()=>{
+        resolve("Data fetched successfully from first promise")
+    },1000)
+})
+
+const p2= new Promise((resolve, reject) =>{
+    setTimeout(()=>{
+        resolve("Data fetched successfully from second promise")
+    },2000)
+}
+)
+
+const p3= new Promise((resolve, reject) =>{
+    setTimeout(()=>{
+        resolve("Data fetched successfully from third promise")
+    },3000)
+}
+)
+
+const apicalling = async()=>{
+    const [a,b,c] =await Promise.all([p1, p2, p3]);
+
+    console.log(a);
+    console.log(b);
+    console.log(c);
+}
+
+apicalling()
+
+console.log(" ")
+// 8. Create a function that fetches data from multiple APIs in parallel and then performs some operation on the combined data, using async/await.
+
+async function fetchData(urls){
+    try{
+        const res = await Promise.all(urls.map(url => fetch(url)));
+
+        const data = await Promise.all(res.map(val => val.json()));
+
+        const combinedData = data.reduce((acc,curr)=>{
+            return {...acc,...curr}
```

Comment:
@anantmishra-successive 
this will overwrite the first api's response data. The ask is to use any operation, which is why this is okay. But I hope you understand what this operation will do.